torch/csrc/jit/python/pybind_utils.h

```diff
@@ -345,6 +345,17 @@ inline InferredType tryToInferType(py::handle input) {
 #endif
   }
 
+  if (as_module(py::cast<py::object>(input))) {
+    // if obj is already a ScriptModule, just return its ivalue
+    return InferredType("Cannot infer type of ScriptModule");
+  }
```

Comment:
This is a bit weird. We can obviously infer the type of a `ScriptModule`, but we may not want to do that if it is in a list or a dict, because we don't allow multiple references to the same module. Should this check be moved to `tryToInferContainerType` so that we can still infer the types of individual attributes that happen to be scripted modules correctly?